cmd/infra/ibmcloud_powervs/create.go

```diff
@@ -690,15 +681,32 @@ func (infra *Infra) createCloudConnection(option *CreateInfraOptions, client *in
 	return cloudConnID, nil
 }
 
-func (infra *Infra) setupPowerVsDhcp(option *CreateInfraOptions, session *ibmpisession.IBMPISession) (err error) {
+func (infra *Infra) deleteDhcp(client *instance.IBMPIDhcpClient) {
+	dhcpServers, err := client.GetAll()
+	if err != nil {
+		log.Log.Error(err, "error on deletion")
+	}
+	for _, dhcp := range dhcpServers {
+		err = client.Delete(*dhcp.ID)
+		if err != nil {
+			log.Log.Error(err, "error on deletion")
+		}
+	}
+}
+
+func (infra *Infra) setupPowerVsDhcp(session *ibmpisession.IBMPISession) (err error) {
 	client := instance.NewIBMPIDhcpClient(context.Background(), session, infra.PowerVSCloudInstanceID)
 
 	var dhcpServer *models.DHCPServerDetail
-	if option.PowerVSDhcpServerID != "" {
-		log.Log.Info("Validating PowerVS DhcpServer", "id", option.PowerVSDhcpServerID)
-		dhcpServer, err = validateDhcpServer(option, client)
-		if err != nil {
-			return err
+
+	dhcpServers, err := client.GetAll()
+	if err != nil {
+		return err
+	}
+	if len(dhcpServers) > 0 {
```

Comment:
add comments why are we doing this here?

Reply:
comments added.